From cb3334999ce938ef511412ca1c97fa6aadac8cde Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Wed, 20 Sep 2017 13:43:11 -0700
Subject: [PATCH] colibri-imx6ull: add configuration for Tezi

Modify configuration for Tezi:
- Enable SDP in U-Boot
- Enable FDT relocation (in-place is causing issues with FIT images)
- Remove Video output
- Use distroboot by default
- Boot from external SD-cards by default
- Disable USB PID from config block (otherwise recovery mode download
  in U-Boot stage fails)

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 configs/colibri-imx6ull_tezi_defconfig | 59 ++++++++++++++++++++++++++++++++++
 include/configs/colibri-imx6ull.h      | 21 ++++++------
 2 files changed, 70 insertions(+), 10 deletions(-)
 create mode 100644 configs/colibri-imx6ull_tezi_defconfig

diff --git a/configs/colibri-imx6ull_tezi_defconfig b/configs/colibri-imx6ull_tezi_defconfig
new file mode 100644
index 0000000000..4aa5d845eb
--- /dev/null
+++ b/configs/colibri-imx6ull_tezi_defconfig
@@ -0,0 +1,59 @@
+CONFIG_ARM=y
+CONFIG_ARCH_MX6=y
+# CONFIG_TDX_CFG_BLOCK_USB_GADGET_PID is not set
+CONFIG_TARGET_COLIBRI_IMX6ULL=y
+CONFIG_DM_SERIAL=y
+CONFIG_DEFAULT_DEVICE_TREE="imx6ull-colibri"
+CONFIG_FIT=y
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/toradex/colibri-imx6ull/imximage.cfg,MX6ULL,IMX_NAND"
+CONFIG_BOOTDELAY=1
+CONFIG_SYS_CONSOLE_IS_IN_ENV=y
+# CONFIG_DISPLAY_BOARDINFO is not set
+CONFIG_HUSH_PARSER=y
+CONFIG_SYS_PROMPT="Colibri iMX6ULL # "
+CONFIG_CMD_BOOTZ=y
+# CONFIG_CMD_IMI is not set
+# CONFIG_CMD_IMLS is not set
+# CONFIG_CMD_XIMG is not set
+CONFIG_CMD_ASKENV=y
+CONFIG_CMD_MEMTEST=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_NAND=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_DFU=y
+CONFIG_CMD_USB_SDP=y
+CONFIG_CMD_USB_MASS_STORAGE=y
+# CONFIG_CMD_FPGA is not set
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_MII=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_CMD_UBI=y
+CONFIG_OF_CONTROL=y
+CONFIG_DFU_MMC=y
+CONFIG_DM_GPIO=y
+CONFIG_DM_I2C=y
+CONFIG_MTD_UBI_FASTMAP=y
+CONFIG_PINCTRL=y
+CONFIG_PINCTRL_IMX6=y
+CONFIG_DM_REGULATOR=y
+# CONFIG_SPL_SERIAL_PRESENT is not set
+CONFIG_MXC_UART=y
+CONFIG_DM_SPI=y
+CONFIG_USB=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_STORAGE=y
+CONFIG_USB_GADGET=y
+CONFIG_CI_UDC=y
+CONFIG_USB_GADGET_DOWNLOAD=y
+CONFIG_G_DNL_MANUFACTURER="Toradex"
+CONFIG_G_DNL_VENDOR_NUM=0x1b67
+CONFIG_G_DNL_PRODUCT_NUM=0x4000
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_FDT_FIXUP_PARTITIONS=y
+# CONFIG_EFI_LOADER is not set
diff --git a/include/configs/colibri-imx6ull.h b/include/configs/colibri-imx6ull.h
index 1e5bae3e9e..696bf0f4e2 100644
--- a/include/configs/colibri-imx6ull.h
+++ b/include/configs/colibri-imx6ull.h
@@ -67,7 +67,6 @@
 #define MEM_LAYOUT_ENV_SETTINGS \
 	"bootm_size=0x10000000\0" \
 	"fdt_addr_r=0x82000000\0" \
-	"fdt_high=0xffffffff\0" \
 	"initrd_high=0xffffffff\0" \
 	"kernel_addr_r=0x81000000\0" \
 	"pxefile_addr_r=0x87100000\0" \
@@ -95,15 +94,15 @@
 	"ubiargs=ubi.mtd=ubi root=ubi0:rootfs rw rootfstype=ubifs " \
 		"ubi.fm_autoconvert=1\0" \
 	"ubiboot=run setup; " \
-		"setenv bootargs ${defargs} ${ubiargs} " \
-		"${setupargs} ${vidargs}; echo Booting from NAND...; " \
-		"ubi part ubi &&" \
-		"ubi read ${kernel_addr_r} kernel && " \
-		"ubi read ${fdt_addr_r} dtb && " \
-		"run fdt_fixup && bootz ${kernel_addr_r} - ${fdt_addr_r}\0" \
+		"setenv bootargs console=ttymxc0,115200 quiet " \
+		"rootfstype=squashfs root=/dev/ram autoinstall " \
+		"${teziargs}; echo Booting Toradex Easy Installer...; " \
+		"ubi part ubi && " \
+		"ubi read ${ramdisk_addr_r} rootfs && " \
+		"bootm ${ramdisk_addr_r}\0" \
 
-#define CONFIG_BOOTCOMMAND "run ubiboot; " \
-	"setenv fdtfile ${soc}-colibri-${fdt_board}.dtb && run distro_bootcmd;"
+#define CONFIG_BOOTCOMMAND "setenv fdtfile ${soc}-colibri-${fdt_board}.dtb && " \
+	"run bootcmd_mmc0; run ubiboot; run distro_bootcmd"
 
 #define BOOTENV_RUN_NET_USB_START ""
 #define BOOT_TARGET_DEVICES(func) \
@@ -172,12 +171,14 @@
 
 /* FLASH and environment organization */
 #define CONFIG_SYS_NO_FLASH
-#define CONFIG_ENV_IS_IN_NAND
+#define CONFIG_ENV_IS_NOWHERE
 
 #if defined(CONFIG_ENV_IS_IN_NAND)
 #define CONFIG_ENV_SECT_SIZE		(128 * 1024)
 #define CONFIG_ENV_OFFSET		(28 * CONFIG_ENV_SECT_SIZE)
 #define CONFIG_ENV_SIZE			CONFIG_ENV_SECT_SIZE
+#else
+#define CONFIG_ENV_SIZE			(8 * 1024)
 #endif
 
 #define CONFIG_NAND_MXS
-- 
2.14.1

